### ISSUE TITLE
Image File URI does not converted in IOS 

### ISSUE THREAD

**Issue by busrauman** [hazemhagrass/phonegap-base64#20]
I'm trying to convert from image file_uri to base64 but xcode return an error : 
errors encountered while discovering extensions: Error Domain=PlugInKit Code=13 "query cancelled" UserInfo={NSLocalizedDescription=query cancelled}

**Comment by tejaskutal** [comment]
Are you fetching the image through Camera/Gallery ?

**Comment by busrauman** [comment]
I'm fetching from camera and gallery with FILE_URI
like that,
 this.camera.getPicture({
           quality: 100,
           destinationType: this.camera.DestinationType.FILE_URI,
           sourceType: this.camera.PictureSourceType.PHOTOLIBRARY,
           encodingType: this.camera.EncodingType.JPEG,
           correctOrientation:true
         }).then(imageData => {
           console.log(imageData)
          }, error => {
           // this.error = JSON.stringify(error);
           console.log(error);
         });

**Comment by tejaskutal** [comment]
You don't need the Base64 plugin then, you can simply change the below line from:
`destinationType: this.camera.DestinationType.FILE_URI,`
to:
`destinationType: this.camera.DestinationType.DATA_URL,`

This will give you the Base64 of the selected/clicked image directly.

**Comment by busrauman** [comment]
yes i know, but i dont want to show image only , i need to upload remote server so i need file_uri
i am using now normalizeUrl(path) to show selected image  it works now, but i want to know why it doesnt work

**Comment by tejaskutal** [comment]
Coz it's been in 'beta' stage since like forever :/

**Comment by busrauman** [comment]
thank you for your advice and time :)

**Comment by tejaskutal** [comment]
Anytime. In case you find a solution to get base64 from the image URI, please post it here. I need it too :)

**Comment by busrauman** [comment]
Ofcourse, if i find anything i will share here:)

**Comment by austinhutchison** [comment]
It's working for me, but I'm using the `cordova-camera-plugin-with-exif` plugin. 

 I use 
`.getPicture({destinationType,1}).then(imageData...`,  

then `normalizeURL(imageData['filename'])` 

then `this.base64.encodeFile(normalizedPath)`
